dali/test/dali_test.h

```diff
@@ -48,17 +54,39 @@ struct Gray {
   static const DALIImageType type = DALI_GRAY;
 };
 
+template <typename T>
+void MeanStdDev(const vector<T> &diff, double *mean, double *std) {
+  const size_t N = diff.size();
+  // Avoid division by zero
+  ASSERT_NE(N, 0);
+
+  double sum = 0, var_sum = 0;
+  for (auto &val : diff) {
+    sum += val;
+  }
+  *mean = sum / N;
+  for (auto &val : diff) {
+    var_sum += (val - *mean)*(val - *mean);
+  }
+  *std = sqrt(var_sum / N);
+}
+
+template <typename T>
+void MeanStdDevColorNorm(const vector<T> &diff, double *mean, double *std) {
+  MeanStdDev(diff, mean, std);
+  *mean /= (255. / 100.);    // normalizing to the color range and use percents
+}
+
 // Main testing fixture to provide common functionality across tests
 class DALITest : public ::testing::Test {
  public:
   inline void SetUp() override {
     rand_gen_.seed(time(nullptr));
-    LoadJPEGS(image_folder, &jpeg_names_, &jpegs_);
-    LoadImages(image_folder, &png_names_, &png_);
+    imageDecrs_[t_jpegImgType].LoadImages(image_folder);
```

Comment:
Would it be beneficial to decode the images once in the SetUp instead when the DecodedBatch is requested?

Reply:
I see that we do this in a subclass but not here.

Reply:
Generally speaking,  Loading and Decoding of the images are two separate process. For some tests (`nvjpegDecodeTest, HostDecodeTest, JpegDecodeTest`) the decoding is done in pipeline by corresponding operator(s). But when the particular test is not testing the decoding process/results, we are calling decoding methods from DALISingleOpTest::SetUp()